### PR TITLE
Introduce Product Page Checkout Button

### PR DESCRIPTION
**Pull request by var-bolt**
This PR attempts to modify the boiler plate and template code we have for Product page checkout in SFCC:
- Introduces the PPC button in `addToCartButtonExtension` template instead of overriding the entire `productDetails` template. 
- Introduces `boltProductPage` button component and JS
- Removes PPC button references from the cart page, cart controller, and the standard bolt button component
- Slightly modifies product controller.

End result:
- When PPC setting is enabled from the business manager the product details page will now include `BoltConnect.js` and then also setup a `div` container for the PPC button. In addition when the button is ready and connect has loaded it will print that in the browser console.

Next step:
- Call the `BoltConnect.ConfigureProductButton` function when product is ready

### CODE REVIEW

File: cartridges/int_bolt_sfra/cartridge/controllers/Product.js [hunk 1]
```diff
@@ -9,9 +9,15 @@ var BoltPreferences = require('int_bolt_core/cartridge/scripts/services/utils/pr
 
 server.append('Show', function (req, res, next) {
     var configuration = BoltPreferences.getSitePreferences();
+    var basketID, hints, dwsid; // eslint-disable-line
```
[the remark this answers is not in the document]
Comment (var-bolt) [reply cartridges/int_bolt_sfra/cartridge/controllers/Product.js]: Removed all unused variables

File: cartridges/int_bolt_sfra/cartridge/client/default/js/boltProductPageButton.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+// initProductPageCheckoutButton Polls:
+// 1) the DOM for the product page checkout button container
+// 2) BoltConnect
+// when they become available, it attaches an onclick event handler to it.
+var initProductPageCheckoutButton = function () {
+    var boltButtonExist = setInterval(function () {
+        var checkoutBoltButton = $('.bolt-product-checkout-button')[0]; // @ts-ignore
+        // have to check if child of this dom is svg, otherwise bolt button is not fully rendered (it's the object)
+        var boltButtonLoaded = checkoutBoltButton && window.BoltCheckout;
```
Comment (serenayanbolt) [inline cartridges/int_bolt_sfra/cartridge/client/default/js/boltProductPageButton.js:11]: Do we need to check if the child is `svg` here?

**Review by serenayanbolt — APPROVED** [review]
LGTM! just left one question

File: cartridges/int_bolt_sfra/cartridge/client/default/js/boltProductPageButton.js (hunk 2 above)
Comment (var-bolt) [reply cartridges/int_bolt_sfra/cartridge/client/default/js/boltProductPageButton.js:11]: You are correct! But I am going to add it in the next PR

**Pull request merged**